lib/src/merged_tree.rs

```diff
@@ -611,6 +614,14 @@ impl TreeMerger {
                     self.mark_completed(&path, value);
                 }
             }
+
+            while self.work.len() < self.store.concurrency() {
+                if let Some(work) = self.unstarted_work.pop_front() {
```

Comment:
It might be better to reorder unstarted_work in a way that earlier (or deeper) nodes will complete early.